### PR TITLE
Simplify balrog implementation

### LINKED ISSUE #797: FVP-02-002 WP1: Balrog does not verify certificate chain on macOS
It was found that Balrog does not verify the whole certificate chain on macOS. This
allows attackers to supply a self-signed leaf certificate, effectively indicating a bypass of
Balrog. This could be abused by state-funded attackers who are in charge of a trusted
valid certificate authority. They could perform a Man-in-the-Middle attack and replace the
binary code provided by the Mozilla VPN update with malicious malware.

### PR DESCRIPTION
**Pull request by birdsarah**
Fixes #797, #798

Instead of rolling our own, we use the new verify module of the mozilla go package autograph (https://github.com/mozilla-services/autograph/).

This allows for simplification of both the api.go code and the update/balrog.cpp. 

In addition, we also resolve the FVP-02-002 and FVP-02-003.

Note, we have verified that balrog algorithm is p384ecdsa which is supported by autograph module.

Thanks to @g-k for splitting out the verify module.

### CODE REVIEW

**Comment by birdsarah** [comment]
@g-k would appreciate a quick review of our balrog/* files to make sure I've used your new module correctly.

File: balrog/api.go [hunk 1]
```diff
@@ -13,220 +13,50 @@ package main
...
+//export balrogValidate
+func balrogValidate(x5u string, input string, signature string, rootHash string, leafCertSubject string) bool {
+	logger := log.New(&CLogger{level: 0}, "", 0)
+	err := contentsignature.Verify([]byte(input), []byte(x5u), signature, rootHash)
+	if err != nil {
+		logger.Println("Verification failed with error:", err.Error())
+		return false
+	}
+	// Verify method does not verify the leaf cert subject, so we do it here.
+	certs, _ := contentsignature.ParseChain([]byte(x5u))
```
Comment (birdsarah) [inline balrog/api.go:53]: ignoring the error because `contentsignature.Verify` already calls ParseChain.

File: src/update/balrog.cpp [hunk 2]
```diff
@@ -174,111 +171,42 @@ bool Balrog::fetchSignature(NetworkRequest* initialRequest,
           });
 
   connect(x5uRequest, &NetworkRequest::requestCompleted,
-          [this, signatureBlob, algorithm, dataUpdate](const QByteArray& data) {
+          [this, signatureBlob, updateData](const QByteArray& x5uData) {
```
Comment (birdsarah) [inline src/update/balrog.cpp:174]: @bakulf as we discussed we leave the network request handling on cpp side.

**Review by bakulf — APPROVED** [review]

**Review by g-k — COMMENTED** [review]
lgtm

**Pull request merged**